portfolio/src/main/webapp/script-home.js

```diff
@@ -33,38 +33,78 @@ window.onload = function() { main(); }
  * function main() initializes the slideshows and the interactive elements on the website.
  */
 function main() {
-    const /** ?HTMLCollection */ slideShowGallery =
-        new SlideShow(document.getElementsByClassName('gallery-slides'));
-    const /** ?HTMLCollection */ slideShowBlog  =
-        new SlideShow(document.getElementsByClassName('blog-slides'));
+    initializeSlideshows();
+    populateComments();
+}
 
-    slideShowBlog.setToAutomaticallyChangeSlides();
+/** 
+ * function initializeSlideshows() initializes the blog slideshow and the gallery slideshow.
+ */
+function initializeSlideshows() {
+  const /** ?HTMLCollection */ slideshowGallery =
+      new SlideShow(document.getElementsByClassName('gallery-slides'));
+  const /** ?HTMLCollection */ slideshowBlog  =
+      new SlideShow(document.getElementsByClassName('blog-slides'));
+  slideshowBlog.setToAutomaticallyChangeSlides();
 
-    document.getElementById('switch-slides-left').onclick =
+  document.getElementById('switch-slides-left').onclick =
         function adjustBackOne() {
           slideShowGallery.adjustSlideManual(ADJUST_BACK); 
-    }
-    document.getElementById('switch-slides-right').onclick =
+        };
+  document.getElementById('switch-slides-right').onclick =
         function adjustForwardOne() {
           slideShowGallery.adjustSlideManual(ADJUST_FORWARD); 
-    }
-    // Fetches the pre-made comment list from the servlet and makes 
-    // a list for each comment.
-    fetch('/comments').then(response => response.json()).then((comments) => {
-      const /** ?HTMLCollection */commentContainer =
-            document.getElementById('comments-container');
-      comments.forEach(function(comment) {
-          commentContainer.appendChild(createListElement(comment));
-      });
-    });
+        };
+}
+
+/** 
+ * function populateComments() populates the comment board on the webpage.
+ */
+function populateComments() {
+  fetch('/comments').then(response => response.json()).then(
+        (comments) => {
+          const /** ?HTMLCollection */commentContainer =
+              document.getElementById('comments-container');
+
+          comments.forEach(function(comment) {
+              commentContainer.appendChild(makeDiv(comment));
+          });
+        }
+    );
 }
 
 /*
- * @return the <li> element containing text passed in.
- * @param {string} text is what is put into the <li>
+ * @return the <div> element containing all the elements of a comment
+ *    like the name, date made, and what was said.
+ * @param {JSONObject} comment is the JSONObject that is
+ *    meant to be turned into a comment
  */
-function createListElement(text) {
-  const liElement = document.createElement('li');
-  liElement.innerText = text;
-  return liElement;
+function makeDiv(comment){
+  let /** string */ author;
```

Comment:
According to the interwebs, you can simply do:

let author = comment.name ? comment.name : 'Anonymous';

https://stackoverflow.com/questions/154059/how-can-i-check-for-an-empty-undefined-null-string-in-javascript